tests/phpunit/tests/functions/TestConfigWpHome.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/**
+ * Test the _config_wp_home function
+ *
+ * @group Functions
+ * @covers ::_config_wp_home
+ */
+class Test_for_config_wp_home extends WP_UnitTestCase {
+
+	/**
+	 * @ticket 57179
+	 * @dataProvider _config_wp_home_dataset
+	 */
+	function test_config_wp_home( $url, $expected ) {
+		$this->assertSame( $expected, _config_wp_home( $url ) );
+	}
+
+	function _config_wp_home_dataset() {
+		return array(
+			'default'   => array(
+				'url'      => '/',
+				'expected' => '/',
+			),
+			'htpps'     => array(
+				'url'      => 'https://www.example.com/',
+				'expected' => 'https://www.example.com/',
+			),
+			'blank_url' => array(
+				'url'      => '',
+				'expected' => '',
+			),
+		);
+	}
+
+	/**
+	 * @ticket 57179
+	 */
+	public function test_get_home_url_from_defined() {
+		if ( ! defined( 'WP_HOME' ) ) {
+			define( 'WP_HOME', 'defined_home' );
+		}
```

Comment:
Unfortunately, as `WP_HOME` is used in several places throughout the codebase, defining it in these tests can cause failures elsewhere in the tests, including in any future tests that might cover code that uses this constant, so the three test methods that define `WP_HOME` can't be added to this test class.

Reply:
So how do we test that if the value is set, we pass the value back to the calling filter?

Reply:
Can we change the function to add a filter for the defined value!!

Reply:
I don't believe we've found consensus in the project for how to resolve this issue.

- Adding a filter would open the `_config_wp_home()` function to extension by plugins.
- We can't run tests in separate processes as PHPUnit allows, to due errors in the WP test suite when attempting to do so.
- We could separate these tests into a different group and exclude this from the test suite (like `ajax`, `ms-files`, `external-http`), but that would lead to many, many excluded groups if this was to be replicated for all testing of constants, and extra CI workflows to run each group in isolation.

@jrfnl Might have some additional context on this.